spec/support/helpers.rb

```diff
@@ -20,8 +20,14 @@ def disabled?(selector)
 
   # click on `selector element when enabled
   def click_on_when_enabled(selector)
-    # will wait until it becomes enabled
-    find("#{selector}:not([disabled])", match: :first).click
+    wait_until { have_css("#{selector}:not([disabled])") }
+    find(selector, match: :first).click
+  end
+
+  def wait_until_having_content(text)
+    has_content = false
+    wait_until { has_content = have_content(text) }
```

Comment:
Yeah, you are right. Need to use the `page.has_text?` as before. Totally forgot.